astropy/utils/masked/tests/test_masked.py

```diff
@@ -295,40 +295,42 @@ def setup_class(self):
         super().setup_class()
         # Ensure we have used MaskedQuantity before - just in case a single test gets
         # called; see gh-15316.
-        Masked(Quantity)
+        self.MQ = Masked(Quantity)
+
+    def test_masked_quantity_getting(self):
+        # First check setup_class (or previous use) defined a cache entry.
+        mcls = Masked._masked_classes[type(self.a)]
+        # Next check this is what one gets now.
+        MQ = Masked(Quantity)
```

Comment:
Why does this line not use `self.MQ` like the other tests? I must have missed some earlier discussions?

Reply:
Yes, here I want to test that if I try to create `MQ` again that I don't get a new version (e.g., in principle, the caching could be broken, and every time one calls it, it creates a new class and just caches that one).